jcg_testadapter_commons/src/main/scala/DOTGraphParser.scala

```diff
@@ -0,0 +1,42 @@
+import java.io.File
+import scala.io.Source
+import scala.util.Using
+
+object DOTGraphParser {
+    /**
+     * Parses a DOT file and returns the nodes and edges
+     *
+     * @param file       the file to parse
+     * @param outputFile the file to write the output to
+     * @return a tuple containing the nodes and edges
+     */
+    def parseFile(file: File): (List[String], List[String]) = {
```

Comment:
Maybe use a more structured type than a tuple of lists of strings, e.g., a specific case class, to make intents clear.